src/redux/sagas/colony/colonyCreate.ts

```diff
@@ -482,13 +486,16 @@ function* colonyCreate({
         [ColonyRole.Architecture, ColonyRole.Root],
       );
 
+      const extensionConfig = supportedExtensionsConfig.find(
+        (config) => config.extensionId === Extension.OneTxPayment,
+      );
       yield put(
         transactionAddParams(setOneTxRoles.id, [
           permissionDomainId,
           childSkillIndex,
           oneTxPaymentExtension.address,
           Id.RootDomain,
-          colonyRoles2Hex([ColonyRole.Administration, ColonyRole.Funding]),
+          colonyRoles2Hex(extensionConfig?.neededColonyPermissions ?? []),
```

Comment:
I like this, however this makes `neededColonyPermissions` to only have a certain format always, otherwise this saga breaks. So we'll have to ensure that it's format never changes.